### PR TITLE
DbMigrator

### PR DESCRIPTION
**Pull request by sorja**

### CODE REVIEW

**Comment by minotogna** [comment]
Task linked: [CU-g54mjp DBMigrator](https://app.clickup.com/t/g54mjp)

File: src/db/index.ts [hunk 1]
```diff
@@ -1,51 +1,3 @@
...
-      user: ProcessEnv.pgUser,
-      database: ProcessEnv.pgDatabase,
-      password: ProcessEnv.pgPassword,
-      host: ProcessEnv.pgHost,
-      port: ProcessEnv.pgPort,
-      ...configCommon,
-    }
-
-export const DB = pgp(config)
+export { migrateAll, migrateSurveySchemas, migrateSurveySchema } from './dbMigrator'
```
Comment (minotogna) [inline src/db/index.ts:3]: see task. DBMigrator should be exported as object (like DB)

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by SteRiccio — APPROVED** [review]

File: src/db/dbMigrator.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    config: { ...config },
+    cwd: `${path.join(__dirname, migrationsFolder)}`,
+    env,
+
+    // Required to work around an EventEmitter leak bug.
+    // See: https://github.com/db-migrate/node-db-migrate/issues/421
+    throwUncatched: true,
+  }
+
+  migrateOptions.config.schema = schema
```
Comment (SteRiccio) [inline src/db/dbMigrator.ts:44]: this can go to line 35

File: src/db/dbMigrator.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+export const migrateSurveySchemas = async () => {
+  const surveyIds = await SurveyService.fetchAllSurveyIds()
+
+  logger.info(`starting data schemas migrations for ${surveyIds.length} surveys`)
+
+  for (const element of surveyIds) {
+    await migrateSurveySchema(element)
+  }
```
Comment (SteRiccio) [inline src/db/dbMigrator.ts:71]: will eslint complain about this? there is an await inside a for loop.
Can this be done in a Promise.all?!

**Review by SteRiccio — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Review by SteRiccio — APPROVED** [review]

**Review by ramirobg94 — APPROVED** [review]

File: src/db/dbMigrator/dbMigrator.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
+import * as path from 'path'
+// @ts-ignore
```
Comment (sorja) [inline src/db/dbMigrator/dbMigrator.ts:2]: is this line still needed?

Comment (minotogna) [reply src/db/dbMigrator/dbMigrator.ts:2]: Yes , you added that actually... not sure why it doesn’t work

**Pull request merged**